xsensemulator/emulator.go

```diff
@@ -100,23 +101,25 @@ func (e *Emulator) SetSendMode() {
 
 func (e *Emulator) Receive(ctx context.Context) error {
 	for {
-		deadline, ok := ctx.Deadline()
-		if !ok {
-			return fmt.Errorf("no deadline")
-		}
-		if err := e.port.SetReadDeadline(deadline); err != nil {
-			return fmt.Errorf("xsens client: receive: %w", err)
+		// Give a chance to quit upon context cancellation
```

Comment:
I guess we're regressing a bit in functionality here, since we're not setting a deadline on the connection anymore. Scan (line 111) would call Read (line 46) on the UDP conn, which would attempt to read indefinitely (without a read timeout) making this context cancellation not trigger and us being stuck here.

Reply:
I might be wrong but I guess its fairly easy to write a test that just doesn't write anything on the UDP conn, but attempts to set a deadline on the context and see if we're returning (I suspect we don't)